app/src/state_machine.cpp

```diff
@@ -20,18 +20,12 @@
 #include <stdbool.h>
 #include "thermostat.hpp"
 #include "driver/gpio.h"
-#define HIGH 1
-#define LOW 0
 
 OPERATING_PARAMETERS OperatingParameters;
 extern int64_t lastTimeUpdate;
 int64_t lastWifiReconnect;
-#ifdef MQTT_ENABLED
-int64_t lastMqttReconnect = 0;
-#endif
 static bool MqttConnectCalled = false;
```

Comment:
@aquini The only for MqttConnectCalled to exist is because I was working thru (and testing) the idea of doing the MQTT initialization once. Some of the func naming in the MQTT module is a little inconsistent, but I was thinking of creating an MqttInit() func I'd call during startup (in main.c).

After a bunch of prototyping and hacking, it's now working well. If after looking at this code path you think there's a better way of doing it, feel free to change it!

Reply:
So, the idea would be to check for the MQTT connection and try to reconnect in a similar way it's done for the WiFi connection, or  MqttConnect() really wants/needs to be called just once?

Reply:
Called just once.

The underlying MQTT implementation automatically reconnects if the connection to the broker (server) is lost. So it's best to let it handle the reconnect.